ui/tests/integration/components/replication-dashboard-test.js

```diff
@@ -0,0 +1,56 @@
+import { module, test } from 'qunit';
+import { setupRenderingTest } from 'ember-qunit';
+import { render } from '@ember/test-helpers';
+import hbs from 'htmlbars-inline-precompile';
+
+const REPLICATION_DETAILS = {
+  state: 'stream-wals',
+  primaryClusterAddr: 'https://127.0.0.1:8201',
+};
+
+const REPLICATION_DETAILS_SYNCING = {
+  state: 'merkle-diff',
+  primaryClusterAddr: 'https://127.0.0.1:8201',
+};
+
+module('Integration | Enterprise | Component | replication-dashboard', function(hooks) {
+  setupRenderingTest(hooks);
+
+  hooks.beforeEach(function() {
+    this.set('replicationDetails', REPLICATION_DETAILS);
+    this.set('replicationDetailsSyncing', REPLICATION_DETAILS_SYNCING);
+    this.set('componentToRender', 'replication-secondary-card');
+    this.set('clusterMode', 'secondary');
+    this.set('isSecondary', true);
+  });
+
+  test('it renders', async function(assert) {
+    await render(hbs`<ReplicationDashboard
+    @replicationDetails={{replicationDetails}}
+    @clusterMode={{clusterMode}}
+    @isSecondary={{isSecondary}}
+    />`);
+
+    assert.dom('[data-test-replication-dashboard]').exists();
+    assert.dom('[data-test-table-rows').exists();
+    assert
+      .dom('[data-test-primary-cluster-address]')
+      .includesText(
+        REPLICATION_DETAILS.primaryClusterAddr,
+        `shows the correct primary cluster address value`
+      );
+    assert.dom('[data-test-replication-doc-link]').exists();
+    assert.dom('[data-test-flash-message]').doesNotExist('no flash message is displayed on render');
+  });
+
+  test('it renders alert banner if state is merkle-diff and isSecondary', async function(assert) {
```

Comment:
whaddya think about making this test description less implementation-specific and more oriented towards what the users sees? i.e. "it renders an alert banner if the dashboard is syncing".

this way if we change our definition of what 'syncing' means (perhaps a primary dashboard will have a syncing state or maybe we decide that something other than `merkle-diff` indicates syncing), this test description is still accurate.

Reply:
Makes a lot of sense.  I'll amend.